numba/array_analysis.py

```diff
@@ -873,6 +901,7 @@ def __init__(self, slice_size, dim_size):
         self.dim_size = dim_size
 
 class ArrayAnalysis(object):
+    aa_count = 0
```

Comment:
Could this be an instance variable instead of a class variable? I'm not sure from the usage why it needs to be scoped like this (IIRC it was a global in the original patch?).

Reply:
I think this should be a class variable instead of an instance variable.  The idea is to number each ArrayAnalysis and to display the number of the array analysis at start and finish.  It isn't always clear where the finish is because there can be nested array analyses.  Those nested analyses don't (always or necessarily) use the same ArrayAnalysis instance (there seem to be two at the moment).  Maybe in the future we could try to make this a singleton class and then the count could be an instance variable.

Reply:
Ah, I see, thanks for the explanation, this makes sense.